app/models/alchemy/page/page_natures.rb

```diff
@@ -101,18 +101,23 @@ def layout_partial_name
         page_layout.parameterize.underscore
       end
 
-      # Returns the version that's taken for Rails' recycable cache key.
+      # Returns the string version of the +last_modified_at+ timestamp.
       #
-      # Uses the +published_at+ value that's updated when the user publishes the page.
+      def cache_version
+        last_modified_at.to_s
+      end
+
+      # Returns the timestamp that the page was last modified at, regardless of through.
+      # publishing or editing page, or through a change of related objects through ingredients.
+      # Respects the public version not changing if editing a preview.
       #
-      # If the page is the current preview it uses the +updated_at+ value as cache key.
+      # In preview mode, it will take the draft version's updated_at timestamp.
+      # In public mode, it will take the public version's updated_at timestamp.
+      # If no version is available, it will take the page's updated_at timestamp.
       #
-      def cache_version
-        if Current.preview_page == self
-          updated_at.to_s
-        else
-          published_at.to_s
-        end
+      def last_modified_at
+        relevant_page_version = (Current.preview_page == self) ? draft_version : public_version
+        relevant_page_version&.updated_at || updated_at
```

Comment:
we always have either a draft or public version and Rails ensures that we always have an updated_at value. we can remove the fallback to the page here.

Reply:
done